jerry-core/lit/lit-strings.c

```diff
@@ -31,89 +31,53 @@ lit_is_valid_utf8_string (const lit_utf8_byte_t *utf8_buf_p, /**< utf-8 string *
                           lit_utf8_size_t buf_size, /**< string size */
                           bool is_strict) /**< true if surrogate pairs are not allowed */
 {
-  lit_utf8_size_t idx = 0;
+  if (!utf8_buf_p)
+  {
+    return false;
+  }
 
-  bool is_prev_code_point_high_surrogate = false;
-  while (idx < buf_size)
+  if (buf_size == 0)
   {
-    lit_utf8_byte_t c = utf8_buf_p[idx++];
-    if ((c & LIT_UTF8_1_BYTE_MASK) == LIT_UTF8_1_BYTE_MARKER)
-    {
-      is_prev_code_point_high_surrogate = false;
-      continue;
-    }
+    return true;
+  }
 
-    lit_code_point_t code_point = 0;
-    lit_code_point_t min_code_point = 0;
-    lit_utf8_size_t extra_bytes_count;
-    if ((c & LIT_UTF8_2_BYTE_MASK) == LIT_UTF8_2_BYTE_MARKER)
+  lit_utf8_size_t extra_bytes_count;
+
+  for (const unsigned char *idx = (const unsigned char *) utf8_buf_p; (unsigned int) (idx - utf8_buf_p) < buf_size;
```

Comment:
It is good to compute the `end` at the beginning and use a single variable. Furthermore the for loop makes the `buf_size == 0` check unnecessary. The 3 parts should go into one line or three lines.